sql/core/src/main/scala/org/apache/spark/sql/execution/aggregate/HashAggregateExec.scala

```diff
@@ -661,6 +661,15 @@ case class HashAggregateExec(
          """.stripMargin
     }
 
+    ctx.addNewFunction("releaseResource", s"""
```

Comment:
We do need to release memory by calling these functions. Because we stop iterating the data early by letting `shouldStop` return true, we won't call next `processNext` and memory leak will happen. So we wrap these calls in override `releaseResource` function.